### PR TITLE
ODC-5545-cypress cucumber helm setup

### PR DESCRIPTION
**Pull request by makambalaji**
**problem**: cypress cucumber helm setup
**solution**: setup the cypress cucumber in helm-plugin folder

JIRA Id: https://issues.redhat.com/browse/ODC-5545 

**Execution**:
Add the below line to pacakage.json
"test-cypress-helm": "cd packages/helm-plugin/integration-tests && ../../../node_modules/.bin/cypress open --env openshift=true",

Execute the command "yarn run test-cypress-helm" and select the "actions-on-helm-release.feature"

**Screenshots**
image

### CODE REVIEW

File: frontend/packages/helm-plugin/integration-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+1. Update the TAGS under env section in config file [Cypress.json file](frontend/packages/pipelines-plugin/integration-tests/cypress.json) as
+   "env": { "TAGS": "@regression and not @manual and not @to-do" }
+2. In command prompt, navigate to frontend folder
+3. Execute command `yarn run test-cypress-pipelines` and select that particular file or run all files in cypress dashboard
+
+Feature file - "regression" suite - execution from command line
+
+1. Open the [frontend/package.json](../../../package.json) and update `test-cypress-pipelines-headless` as per requirement
+2. In command line, navigate to frontend folder and execute the command `yarn run test-cypress-pipelines-headless`
+3. All the regression scenarios get executed as per the configuration.
```
Comment (gajanan-more) [inline frontend/packages/helm-plugin/integration-tests/README.md:75]: It's related to Helm Plugin. There is few stuff related to Pipelines plugin that we should update.

File: frontend/packages/helm-plugin/integration-tests/features/helm/actions-on-helm-release.feature [hunk 2]
```diff
@@ -0,0 +1,59 @@
+@helm
+Feature: Perform Actions on Helm Releases
+              As a user, I want to perform the actions on the helm releases in topology page
+
+        Background:
+              Given user has created or selected namespace "aut-helm"
```
Comment (gajanan-more) [inline frontend/packages/helm-plugin/integration-tests/features/helm/actions-on-helm-release.feature:6]: No proper indentation on this line.

File: frontend/packages/helm-plugin/integration-tests/features/helm/actions-on-helm-release.feature [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+
+
+        @smoke
+        Scenario: Context menu options of helm release: HR-07-TC01
+            Given helm release "nodejs-ex-k" is present in topology page
```
Comment (gajanan-more) [inline frontend/packages/helm-plugin/integration-tests/features/helm/actions-on-helm-release.feature:11]:
```suggestion
            Given user has installed helm release "nodejs-ex-k"
```

File: frontend/packages/helm-plugin/integration-tests/features/helm/actions-on-helm-release.feature [hunk 4]
```diff
@@ -0,0 +1,59 @@
+@helm
+Feature: Perform Actions on Helm Releases
+              As a user, I want to perform the actions on the helm releases in topology page
```
Comment (gajanan-more) [inline frontend/packages/helm-plugin/integration-tests/features/helm/actions-on-helm-release.feature:3]: Indentation

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-compatibility.feature [hunk 5]
```diff
@@ -0,0 +1,30 @@
+@helm
+Feature: Helm Chart
+    User will be able to update the chart versions or values config of a helm release
+
+
+        Background:
+              Given user has created or selected namespace "aut-helm"
```
Comment (gajanan-more) [inline frontend/packages/helm-plugin/integration-tests/features/helm/helm-compatibility.feature:7]: Indentation

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-feature-flag.feature [hunk 6]
```diff
@@ -0,0 +1,31 @@
+@helm @regression @manual
+Feature: Feature flag for Helm
+              As a user, I want to disable helm specific navigation items from console if there are no helm repositories configured in the console.
```
Comment (gajanan-more) [inline frontend/packages/helm-plugin/integration-tests/features/helm/helm-feature-flag.feature:3]: Indentation

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-feature-flag.feature [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+
+        Background:
+              Given user has created or selected namespace "aut-helm"
```
Comment (gajanan-more) [inline frontend/packages/helm-plugin/integration-tests/features/helm/helm-feature-flag.feature:7]: Indentation

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-feature-flag.feature [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+
+        Scenario: Disable helm features in console
```
Comment (gajanan-more) [inline frontend/packages/helm-plugin/integration-tests/features/helm/helm-feature-flag.feature:10]: We don't have any tag for this scenario. WDYT about adding tag?

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-feature-flag.feature [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+             When user opens "redhat-helm-repo" CR
+              And user goes to YAML tab
+              And user adds "disabled: true" flag under "spec"
+              And user clicks on Save
+             Then user is not able to see the Helm tab in the navigation menu
+              And user can not see Helm Chart card in Add page
+              And user can not see Helm Charts filter in the Developer Catalog page
+
+
+        Scenario: Enable the disabled helm features in console
```
Comment (gajanan-more) [inline frontend/packages/helm-plugin/integration-tests/features/helm/helm-feature-flag.feature:22]: We don't have any tag for this scenario. WDYT about adding tag?

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-installation-view.feature [hunk 10]
```diff
@@ -0,0 +1,38 @@
+@helm
+Feature: Helm Chart Installation View
+              As a user, I should be able switch between YAML and Form view to install Helm Chart
```
Comment (gajanan-more) [inline frontend/packages/helm-plugin/integration-tests/features/helm/helm-installation-view.feature:3]: Indentation

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-navigation.feature [hunk 11]
```diff
@@ -0,0 +1,126 @@
+@helm
+Feature: Navigations on Helm Chart
+              As a user, I want to navigate to different pages related to Helm Charts
```
Comment (gajanan-more) [inline frontend/packages/helm-plugin/integration-tests/features/helm/helm-navigation.feature:3]: Indentation

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-navigation.feature [hunk 12]
```diff
@@ -0,0 +1,126 @@
...
+
+        Background:
+              Given user has created or selected namespace "aut-helm"
+
+        @smoke
```
Comment (gajanan-more) [inline frontend/packages/helm-plugin/integration-tests/features/helm/helm-navigation.feature:8]: Can you please add two lines between two scenarios, Background and next scenario.

File: frontend/packages/helm-plugin/integration-tests/features/helm/install-helm-chart.feature [hunk 13]
```diff
@@ -0,0 +1,84 @@
+@helm
+Feature: Install the Helm Release
+              As a user, I want to install the helm release
```
Comment (gajanan-more) [inline frontend/packages/helm-plugin/integration-tests/features/helm/install-helm-chart.feature:3]: Indentation

File: frontend/packages/helm-plugin/integration-tests/features/helm/install-helm-chart.feature [hunk 14]
```diff
@@ -0,0 +1,84 @@
...
+
+
+        @regression @manual
+        Scenario: README should be updated when chart version is updated
+            Given user is at Install Helm Chart page
+             Then user will see the chart version dropdown
+             When user selects YAML view
+             When user selects the Chart Version "0.2.0 / App Version 1.16.0 (Provided by Red Hat Helm Charts)"
+             When user selects "Proceed" button from Change Chart version confirmation dialog
+             Then user will see that the README is also updated with new chart version "0.2.0 / App Version 1.16.0 (Provided by Red Hat Helm Charts)"
```
Comment (gajanan-more) [inline frontend/packages/helm-plugin/integration-tests/features/helm/install-helm-chart.feature:84]: Can you please add new line?

**Review by gajanan-more — CHANGES_REQUESTED** [review]
There are few indentation errors while describing the feature. Can you please have a look at them?

File: frontend/packages/helm-plugin/integration-tests/README.md (hunk 1 above)
Comment (makambalaji) [reply frontend/packages/helm-plugin/integration-tests/README.md:75]: updated

File: frontend/packages/helm-plugin/integration-tests/features/helm/actions-on-helm-release.feature (hunk 3 above)
Comment (makambalaji) [reply frontend/packages/helm-plugin/integration-tests/features/helm/actions-on-helm-release.feature:11]: udpated

File: frontend/packages/helm-plugin/integration-tests/features/helm/actions-on-helm-release.feature (hunk 4 above)
Comment (makambalaji) [reply frontend/packages/helm-plugin/integration-tests/features/helm/actions-on-helm-release.feature:3]: It is automatically updated on save, on gherkin-lint able to see this issue

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-compatibility.feature (hunk 5 above)
Comment (makambalaji) [reply frontend/packages/helm-plugin/integration-tests/features/helm/helm-compatibility.feature:7]: updated

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-installation-view.feature (hunk 10 above)
Comment (makambalaji) [reply frontend/packages/helm-plugin/integration-tests/features/helm/helm-installation-view.feature:3]: Updated automatically on save and not able to see any gherkin-lint issue

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-navigation.feature (hunk 11 above)
Comment (makambalaji) [reply frontend/packages/helm-plugin/integration-tests/features/helm/helm-navigation.feature:3]: Updated automatically on save and not able to see any gherkin-lint issue

File: frontend/packages/helm-plugin/integration-tests/features/helm/helm-navigation.feature (hunk 12 above)
Comment (makambalaji) [reply frontend/packages/helm-plugin/integration-tests/features/helm/helm-navigation.feature:8]: updated

File: frontend/packages/helm-plugin/integration-tests/features/helm/install-helm-chart.feature (hunk 13 above)
Comment (makambalaji) [reply frontend/packages/helm-plugin/integration-tests/features/helm/install-helm-chart.feature:3]: Updated automatically on save and not able to see any gherkin-lint issue

File: frontend/packages/helm-plugin/integration-tests/features/helm/install-helm-chart.feature (hunk 14 above)
Comment (makambalaji) [reply frontend/packages/helm-plugin/integration-tests/features/helm/install-helm-chart.feature:84]: updated

**Review by gajanan-more — APPROVED** [review]
Screenshot from 2021-03-31 10-35-49

Working as expected 

/lgtm

**Pull request merged**